tools/templates/controllers/ApiController.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+namespace YesWiki\Templates\Controller;
+
+use Symfony\Component\Routing\Annotation\Route;
+use YesWiki\Core\ApiResponse;
+use YesWiki\Core\YesWikiController;
+use YesWiki\Core\Service\ThemeManager;
+
+class ApiController extends YesWikiController
+{
+    /**
+     * @Route("/api/templates/custom-presets/{presetFilename}", methods={"DELETE"},options={"acl":{"public","@admins"}})
+     */
+    public function deleteCustomCSSPreset($presetFilename)
+    {
+        return new ApiResponse(['status'=>
+            ($this->getService(ThemeManager::class)->deleteCustomCSSPreset($presetFilename))
+                ? 1 // 'OK'
+                : 0 // 'not OK
+            ]);
+    }
+
+    /**
+     * @Route("/api/templates/custom-presets/{presetFilename}", methods={"POST"},options={"acl":{"public","+"}})
+     */
+    public function addCustomCSSPreset($presetFilename)
+    {
+        $res = $this->getService(ThemeManager::class)->addCustomCSSPreset($presetFilename, $_POST);
+        return new ApiResponse(['status'=>
+            (!empty($res))
+                ? 1 // 'OK'
+                : 0 // 'not OK
```

Comment:
Il vaudrait mieux qu'ApiResponse envoie 200 pour un succes et 404 pour not found en cas d'erreur
```php
if ($res == true) {
  return new ApiResponse(['filename'=> $res], 200);
} else {
  return new ApiResponse(['status' => 404, 'message'=> _t('TEMPLATES_PRESET_NOT_FOUND')], 404);
}
```

Reply:
OK, je fais une nouvelle proposition pour cette partie (légèrement différente, car je pense que les messages de l'api peuvent rester en anglais sans traduction.